src/NuGet.Core/NuGet.ProjectModel/JsonPackageSpecReader.cs

```diff
@@ -806,16 +807,32 @@ private static bool BuildTargetFrameworkNode(PackageSpec packageSpec, KeyValuePa
             return true;
         }
 
-        private static void PopulateFrameworkReferences(ISet<string> frameworkReferences, JObject properties, string frameworkReferenceName)
+        private static void PopulateFrameworkReferences(ISet<FrameworkDependency> frameworkReferences, JObject properties, string frameworkReferenceName, string packageSpecPath)
         {
-            var frameworkRefProperty = properties[frameworkReferenceName];
+            var frameworkRefs = properties[frameworkReferenceName] as JObject;
 
-            if (frameworkRefProperty != null)
+            if (frameworkRefs != null)
             {
-                IEnumerable<string> frameworkRefArray;
-                if (TryGetStringEnumerableFromJArray(frameworkRefProperty, out frameworkRefArray))
+                foreach (var frameworkReference in frameworkRefs)
                 {
-                    frameworkReferences.AddRange(frameworkRefArray);
+                    if (string.IsNullOrEmpty(frameworkReference.Key))
+                    {
+                        throw FileFormatException.Create(
+                                   "Unable to resolve frameworkReference ''.",
+                                   frameworkReference.Value,
+                                   packageSpecPath);
+                    }
+
+                    var privateAssets = FrameworkDependencyFlagsUtils.Default;
+
+                    if (frameworkReference.Value is JObject frameworkReferenceToken)
+                    {
+                        if (TryGetStringEnumerable(frameworkReferenceToken["privateAssets"], out var strings))
```

Comment:
Why is this `privateAssets` but [this](https://github.com/NuGet/NuGet.Client/pull/2829/files#diff-bb46f9dc27153194714455bf3f388f57R64) is `PrivateAssets`?

Reply:
It's a pattern in our package spec.
Inner objects are camelcased.